### PR TITLE
feat: add E-Waste Impact Calculator & Recycling Locator (#2028)

### LINKED ISSUE #2028: 💡[FEATURE]: Adding E-Waste Impact Calculator & Recycling Locator
I would like to propose a new ML-based tool that calculates the environmental footprint of electronic devices and helps users find local recycling centers. With the rising consumption of electronics, managing e-waste has become a critical environmental challenge. This project will enable users to input their device details to assess its carbon footprint and toxic material potential.

Proposed Features:

Impact Calculator: An ML model that estimates the environmental impact based on gadget type, model, and age.

Recycler Locator: A feature to identify the nearest authorized e-waste recycling centers using geospatial data.

Educational Dashboard: A summary report highlighting the importance of responsible disposal.

Why it's unique:
Most ML projects focus on predictive analytics or basic classification. This project integrates Environmental Data Science with Geospatial API mapping, providing users with an actionable way to contribute to a greener planet. It moves beyond just "prediction" and offers a "solution-oriented" approach.

Tech Stack:

1. ML Model: Python (Scikit-learn/Random Forest for impact prediction)

2. Data Handling: Pandas, NumPy

3. UI/Frontend: Streamlit (for a lightweight, web-based dashboard)

4. Integrations: Google Maps API / OpenStreetMap (for recycler locations)

### PR DESCRIPTION
**Pull request by Rakshak05**
Closes #2028    

### Description
This PR introduces a machine learning-based tool to compute the environmental footprint of electronic devices and help users locate nearby authorized recycling centers.

### Key Features Added
1. **Impact Calculator:** 
   - A multi-output Random Forest regressor trained on synthetic lifecycle data.
   - Computes lifecycle **CO₂ Carbon Footprint** (in kg CO₂ eq), **Toxic Material Score** (0-100), and **Recyclability Score** (0-100%).
   - Displays real-world environmental equivalencies (e.g. standard petrol car miles, smartphone charges, tree absorption equivalents) to make the impact tangible.
2. **Recycler Locator:**
   - Computes geospatial distances to authorized recycling facilities using the Haversine formula.
   - Plots disposal centers dynamically on an interactive map.
3. **Educational Dashboard:**
   - Highlights global e-waste stats and details hazards of toxic compounds (e.g. Lead, Mercury).
   - Provides checklist steps to prepare devices before recycling.

### Project Structure
- `E-Waste_Impact_Calculator_Recycling_Locator/train_model.py`: Script generating datasets, training scikit-learn models, and saving the pipeline parameters.
- `E-Waste_Impact_Calculator_Recycling_Locator/app.py`: Streamlit-driven interactive application.
- `E-Waste_Impact_Calculator_Recycling_Locator/requirements.txt`: Specific library packages required for execution.
- Added project row reference to the `ROADMAP.md` under the Intermediate level track.

### Testing/Verification
- Ran `python train_model.py` to verify dataset generation and successful model fitting.
  - CO₂ Footprint Regressor: MAE ~ `13.03` | R² ~ `0.99`
  - Toxic Score Regressor: MAE ~ `4.16` | R² ~ `0.89`
  - Recyclability Regressor: MAE ~ `2.60%` | R² ~ `0.98`
- Tested running `streamlit run app.py` locally and verified correct navigation, distance metrics, and rendering of the Streamlit map.

### CODE REVIEW

File: E-Waste_Impact_Calculator_Recycling_Locator/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+- **Interactive Recycler Locator:** Helps users locate authorized disposal centers using the Haversine distance algorithm, showing results in a localized table and a dynamic map (built via Streamlit's `st.map`).
+- **Educational Dashboard:** Provides actionable guides (data sanitization, battery removal), toxic compound insights, and impact statistics.
+
+## Project Structure
+
+```text
+E-Waste_Impact_Calculator_Recycling_Locator/
+├── train_model.py     # Data generation & Random Forest model training script
+├── app.py             # Streamlit web application
+├── dataset.csv        # Generated synthetic dataset (created after training)
```
Comment (Sibam-Paul) [inline E-Waste_Impact_Calculator_Recycling_Locator/README.md:20]: u need to add the link of the dataset in here

File: E-Waste_Impact_Calculator_Recycling_Locator/train_model.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    recyclability += np.random.normal(0, 3)
+    recyclability = np.clip(recyclability, 5, 98)
+    
+    return pd.Series([round(co2, 2), round(toxic, 1), round(recyclability, 1)])
+
+# Apply the metrics function to generate labels
+df[["co2_footprint", "toxic_score", "recyclability"]] = df.apply(calculate_metrics, axis=1)
+
+# Save generated dataset to CSV for reference
+df.to_csv(os.path.join(BASE_DIR, "dataset.csv"), index=False)
```
Comment (Sibam-Paul) [inline E-Waste_Impact_Calculator_Recycling_Locator/train_model.py:130]: handle the dataset using link 
u can upload the dataset in the github gist and then make a download handle like
```bash
import pandas as pd

# Replace the URL below with your actual raw gist URL
url = "https://githubusercontent.com"

# Load the CSV into a pandas DataFrame
df = pd.read_csv(url)

# Display the first few rows to verify
df.head()
```

don't add hardcoded csv file paths 

if u are using kaggle add handles to download it directly form kaggle in the code base 

@Rakshak05

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]
attach the screenshot also of it's working @Rakshak05

**Comment by Rakshak05** [comment]
app_screenshot

**Review by Sibam-Paul — APPROVED** [review]
LGTM!
